qiskit/utils/mitigation/filters.py

```diff
@@ -0,0 +1,498 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2019.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+# This code was originally copied from the qiskit-ignis see:
+# https://github.com/Qiskit/qiskit-ignis/blob/b91066c72171bcd55a70e6e8993b813ec763cf41/qiskit/ignis/mitigation/measurement/filters.py
+# it was migrated as qiskit-ignis is being deprecated
+
+# pylint: disable=cell-var-from-loop,invalid-name
+
+
+"""
+Measurement correction filters.
+
+"""
+
+from typing import List
+from copy import deepcopy
+
+import numpy as np
+from scipy.optimize import minimize
+import scipy.linalg as la
+
+import qiskit
+from qiskit import QiskitError
+from qiskit.tools import parallel_map
+from qiskit.utils.mitigation.circuits import count_keys
+
+
+class MeasurementFilter:
+    """
+    Measurement error mitigation filter.
+
+    Produced from a measurement calibration fitter and can be applied
+    to data.
+
+    """
+
+    def __init__(self, cal_matrix: np.matrix, state_labels: list):
+        """
+        Initialize a measurement error mitigation filter using the cal_matrix
+        from a measurement calibration fitter.
+
+        Args:
+            cal_matrix: the calibration matrix for applying the correction
+            state_labels: the states for the ordering of the cal matrix
+        """
+
+        self._cal_matrix = cal_matrix
+        self._state_labels = state_labels
+
+    @property
+    def cal_matrix(self):
+        """Return cal_matrix."""
+        return self._cal_matrix
+
+    @property
+    def state_labels(self):
+        """return the state label ordering of the cal matrix"""
+        return self._state_labels
+
+    @state_labels.setter
+    def state_labels(self, new_state_labels):
+        """set the state label ordering of the cal matrix"""
+        self._state_labels = new_state_labels
+
+    @cal_matrix.setter
+    def cal_matrix(self, new_cal_matrix):
+        """Set cal_matrix."""
+        self._cal_matrix = new_cal_matrix
+
+    def apply(self, raw_data, method="least_squares"):
+        """Apply the calibration matrix to results.
+
+        Args:
+            raw_data (dict or list): The data to be corrected. Can be in a number of forms:
+
+                 Form 1: a counts dictionary from results.get_counts
+
+                 Form 2: a list of counts of `length==len(state_labels)`
+
+                 Form 3: a list of counts of `length==M*len(state_labels)` where M is an
+                 integer (e.g. for use with the tomography data)
+
+                 Form 4: a qiskit Result
```

Comment:
Yeah I think we're ok here since we're not going to building these in the docs (it also limits how much cleanup we need to do in the migration here). Fwiw it does look weird in the ignis docs: https://qiskit.org/documentation/stubs/qiskit.ignis.mitigation.MeasurementFilter.apply.html#qiskit.ignis.mitigation.MeasurementFilter.apply but it's not as bad as I was expecting.

Reply:
Actually not that messed up at all, really!  I suppose Sphinx correctly interpreted it as a series of paragraphs, within the same argument, though arguably the intent was probably a bulleted list.